doc/import_export.rst

```diff
@@ -17,78 +14,46 @@ Create a folder with CSVs for each component type
 
-Note that is is NOT necessary to add every single column, only those where values differ from the defaults listed in :doc:`components`. All empty values/columns are filled with the defaults.
+.. note:: Note that is is NOT necessary to add every single column, only those where values differ from the defaults listed in :doc:`components`. All empty values/columns are filled with the defaults.
 
 
 .. _export-csv:
 
 Export to folder of CSV files
 =============================
 
-The network can be exported as a folder of csv files:
-
-``network.export_to_csv_folder(csv_folder_name)``
-
-If the folder does not exist it will be created.
-
-All non-default static and series attributes of all components will be
-exported.
-
-Static attributes are exported in one CSV file per component,
-e.g. ``generators.csv``.
-
-Series attributes are exported in one CSV file per component per
-attribute, e.g. ``generators-p_set.csv``.
-
+The network can be exported as a folder of csv files with ``network.export_to_csv_folder(csv_folder_name)``.
 
-Adding components one-by-one
-============================
+.. autofunction:: pypsa.io.export_to_csv_folder
 
-Networks can also be built "manually" in code by calling
 
-``network.add(class_name, name, **kwargs)``
+Adding and removing components one-by-one
+==========================================
 
-where ``class_name`` is for example
-``"Line","Bus","Generator","StorageUnit`` and ``name`` is the unique
-name of the component. Other attributes can also be specified:
+Networks can also be built step-by-step for each component by calling
 
-.. code:: python
-
-    network.add("Bus","my_bus_0")
-    network.add("Bus","my_bus_1",v_nom=380)
-    network.add("Line","my_line_name",bus0="my_bus_0",bus1="my_bus_1",length=34,r=2,x=4)
-
-Any attributes which are not specified will be given the default value from :doc:`components`.
-
-This method is slow for many components; instead use ``madd`` or
-``import_components_from_dataframe`` (see below).
+.. automethod:: pypsa.Network.add
 
+Likewise, components can also be removed with
 
+.. automethod:: pypsa.Network.remove
 
 .. _madd:
 
-Adding multiple components
-==========================
+Adding and removing multiple components
+========================================
 
 Multiple components can be added by calling
 
-``network.madd(class_name, names, **kwargs)``
+.. automethod:: pypsa.Network.madd
 
-where ``class_name`` is for example
-``"Line","Bus","Generator","StorageUnit`` and ``names`` is a list of
-unique names of the components. Other attributes can also be specified
-as scalars, lists, arrays, pandas Series or pandas DataFrames.
+Multiple components can be removed by calling
 
-Make sure when adding static attributes as pandas Series that they are
-indexed by names. Make sure when adding time-varying attributes as
-pandas DataFrames that their index is a superset of network.snapshots
-and their columns are a subset of names.
+.. automethod:: pypsa.Network.mremove
```

Comment:
I'd move the `mremove` method after the longer python example below or even integrate the example into the `madd` docstring.